samtranslator/model/sam_resources.py

```diff
@@ -867,8 +868,123 @@ def to_cloudformation(self, **kwargs):
+
         return resources
 
+    def _validate_cloudwatch_alarms(self):
+        """Validates the CloudWatchAlarms property in Serverless Canary
+
+        The property should follow the following structure
+
+        CloudWatchAlarms:
+            - AlarmName:
+                MetricName (required): one of ["SuccessPercent", "Failed", "Duration"]
+                Threshold (optional): any value of type double
+                ComparisonOperator (optional): any of the valid values (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-alarm.html#cfn-cloudwatch-alarms-comparisonoperator)
+                Period (optional): Integer that is 10, 30, 60, or any multiple of 60
+            - AlarmName2:
+                MetricName: Failed
+
+        Note: Alarm names are used as logical ids for their respective CloudWatchAlarm property so there should be no
+        duplicate names as we don't want the alarms to override each other without the user knowing
+
+        :raise: InvalidResourceException
+        """
+        # keeps list of alarm names to make sure there are no duplicates
+        list_of_alarm_names = []
+        for dict_item in self.CloudWatchAlarms:
+
+            # Throw an error if there is more than one alarm in the array index, like for example
+            # CloudWatchAlarms:
+            #     - Alarm1:
+            #         MetricName: SuccessPercent
+            #       Alarm2:
+            #         MetricName: SuccessPercent
+            #         Threshold: 90
+            if len(dict_item) > 1:
+                raise InvalidResourceException(self.logical_id, "Must only have one alarm per array index")
+
+            # get the alarm name and the properties the user defined for the alarm
+            alarm_name = list(dict_item.keys())[0]
+            alarm_item = dict_item[alarm_name]
+
+            # MetricName property is required
+            if alarm_item is None or "MetricName" not in alarm_item:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "CloudWatch alarm '{key}' is missing required property 'MetricName'.".format(key=alarm_name),
+                )
+
+            metric_name = alarm_item["MetricName"]
+            valid_metrics = ["SuccessPercent", "Failed", "Duration"]
+
+            # MetricName must be one of "SuccessPercent", "Failed" and "Duration"
+            if metric_name not in valid_metrics:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "MetricName needs to be one of {}".format(valid_metrics),
+                )
+
+            # make sure all the alarm names are unique
+            if alarm_name in list_of_alarm_names:
+                raise InvalidResourceException(self.logical_id, "Duplicate CloudWatch alarm names")
+            else:
+                list_of_alarm_names.append(alarm_name)
+
+    def _construct_cloudwatch_alarms(self, dict_item):
```

Comment:
what exactly is dict_item? It is not even in method documentation. Can we think of a better name?